module-apps/application-call/ApplicationCall.cpp

```diff
@@ -82,34 +82,45 @@ namespace app
             return actionHandled();
         });
         addActionReceiver(manager::actions::AbortCall, [this](auto &&data) {
-            callerIdTimer.reset();
             if (const auto state = getState(); state == Application::State::ACTIVE_FORGROUND) {
-                switchWindow(window::name_call, std::make_unique<app::CallAbortData>());
+                switchWindow(window::name_call);
             }
             else if (state == Application::State::ACTIVE_BACKGROUND) {
-                setCallState(call::State::IDLE);
+                callModel->setState(app::call::CallState::None);
```

Comment:
to completely follow we should set the model first -> then based on presenter reaction switch window instead? I'm not requesting it. There always has to be some limit on how much we want to change code for the better at once and I firmly believe that this PR met the requirements and above :)